### PR TITLE
Reduce size of HyperLogLog struct

### PR DESCRIPTION
**Pull request by imay**
Now size of HyperLogLog struct is so large that it lead the rowset is
too small when ingesting data. In this CL, registers in HyperLogLog are
only created when it is needed. When ingesting data, it's normal case
that there are only few values in one HyperLogLog.

For #1837

### CODE REVIEW

File: be/src/exprs/hll_function.cpp [hunk 1]
```diff
@@ -32,16 +32,16 @@ StringVal HllFunctions::hll_hash(FunctionContext* ctx, const StringVal& input) {
     const int HLL_SINGLE_VALUE_SIZE = 10;
```
Comment (morningman) [inline be/src/exprs/hll_function.cpp]: add a comment to the number `10`? (1+1+8)?

File: be/src/olap/hll.h [hunk 2]
```diff
@@ -23,59 +23,68 @@
...
-// empty 空集合
-// explicit 存储64位hash值的集合
-// sparse 存储hll非0的register
-// full  存储全部的hll register
-// empty -> explicit -> sparse -> full 四种类型的转换方向不可逆
-// 第一个字节存放hll集合的类型 0:empty 1:explicit 2:sparse 3:full
-// 已决定后面的数据怎么解析
+// Each HLL value is a set of values. To save space, Doris store HLL value
+// in different format according to its cardinality.
+//
```
Comment (morningman) [inline be/src/olap/hll.h]: Is there a fix scope for what type should we the choice?
eg.
`1~160` using HLL_DATA_EXPLICIT
`161~4096` using HLL_DATA_SPRASE?

**Review by morningman — COMMENTED** [review]
Nice job!
In 1 GB date loading, segments number reduces from 4k+ to 20+

File: be/src/exprs/hll_function.cpp (hunk 1 above)
Comment (imay) [reply be/src/exprs/hll_function.cpp]: OK

File: be/src/olap/hll.h (hunk 2 above)
Comment (imay) [reply be/src/olap/hll.h]: OK, I will add some comment here

File: be/src/olap/hll.cpp [hunk 3]
```diff
@@ -20,172 +20,236 @@
...
+    // clear _hash_set
+    std::set<uint64_t>().swap(_hash_set);
+}
 
-            for (auto iter: _sparse_map) {
-                _registers[iter.first] = (uint8_t)iter.second;
-            }
-        }
+// Change HLL_DATA_EXPLICIT to HLL_DATA_FULL directly because they are
+// implemented in the same way in memory.
```
[the remark this answers is not in the document]
Comment (imay) [reply be/src/olap/hll.cpp]: Typo, HLL_DATA_SPASE and HLL_DATA_FULL

File: be/src/olap/hll.h (hunk 2 above)
Comment (imay) [reply be/src/olap/hll.h]: OK, I will add some comment here

**Review by kangkaisen — APPROVED** [review]
+1. The logic is more clear!

**Pull request merged**